src/Manager.php

```diff
@@ -125,7 +191,46 @@ public function compare($name, array $config = [])
      */
     public function custom($name, array $config = [])
     {
-        $this->_config[$name] = new $config['className']($name, $this, $config);
+        $this->add($name, $config['className'], $config);
         return $this;
     }
+
+    /**
+     * Loads a search filter.
+     *
+     * @param string $name Name of the field
+     * @param string $filter Filter name
+     * @param array $options Filter options.
+     * @return \Search\Type\Base
+     * @throws \InvalidArgumentException When no filter was found.
+     */
+    public function loadFilter($name, $filter, array $options = [])
+    {
+        if (empty($options['className'])) {
+            $class = Inflector::classify($filter);
+        } else {
+            $class = $options['className'];
+            unset($options['className']);
+        }
+        $className = App::className($class, 'Type');
```

Comment:
Yeah, that's a drawback of this simplification. We could change the namespace for the type classes. For one I would prefer renaming `Type` to `Filter`,  so that we have `src/Filter/Like.php` or better yet we can have `src/ORM/Filter/Like.php`.